README.md

```diff
@@ -1,6 +1,193 @@
+prevent job triggering by users and other jobs using various conditions.
+
+# About the plugin
+
+### Features:
+
+-   Execution restrictions on the node level
+    -   Example 1: Take jobs according to the specified name pattern
+        (e.g. allow running only "QA\_.\*" jobs on a node)
+    -   Example 2: Prevent execution of user jobs on the Jenkins master
+        node
+-   Restrictions of jobs triggering by external causes
+    -   Example 1: Prohibit manual builds
+    -   Example 2: Allow triggering only by a job with the
+        specified [owner](https://wiki.jenkins.io/display/JENKINS/Ownership+Plugin).
+-   Several built-in restrictions + JobRestriction extension point
+
+### Contributors:
+
+1.  [Oleg Nenashev](hhttp://www.linkedin.com/in/onenashev) (maintainer)
+    -   Initial plugin version has been sponsored by Synopsys Inc.,
+        [www.synopsys.com](http://www.synopsys.com/)
+
+# Usage
+
+## Job restrictions for nodes
+
+These restrictions can be configured in node's configuration page.
+
+-   Node won't accept jobs, which violate the specified restrictions
+-   Other requirements (executors, labels, etc.) will be checked as well
+
+![](docs/images/Jenkins_JobRestrictions_NodeRestriction.png)
+
+## Job restrictions for projects
+
+This type of restrictions allows to prevent execution of jobs by the
+launch cause.  
+If the cause does not satisfy requirements, job fails before running of
+SCM (the job cannot be aborted due to
+[JENKINS-19497](https://issues.jenkins-ci.org/browse/JENKINS-19497)).
+
+The current version supports the following checks:
+
+-   **Apply Job restrictions to upstream build** - prohibits the
+    invocation from specific builds, which do not satisfy the specified
+    requirements
+-   Prohibit manual launch of the job (since 0.2)
+
+# Extension points
+
+Plugin provides a JobRestriction extension point, which allows
+implementation of new restrictions.
+
+Built-in extensions:
+
+-   Logic operations (and, or, not)
+-   Started by user/group restrictions
+    -   Restrictions support upstream projects (UpstreamCause)
+    -   [Rebuild
+        Plugin](https://wiki.jenkins.io/display/JENKINS/Rebuild+Plugin)
+        and other specific causes are not supported
+-   Regex restriction - check the jobs name by a regular expression
+
+The following plugins produce additional job restrictions:
+
+-   Page:
+    [Ownership
+    Plugin](https://wiki.jenkins.io/display/JENKINS/Ownership+Plugin)
+
+-   Page:
+    [Priority Sorter
+    Plugin](https://wiki.jenkins.io/display/JENKINS/Priority+Sorter+Plugin)
+    — This plugin allows Jobs to be prioritised based on *Queue
+    Strategies* and *Priority Strategies*.
+
+If your plug-in is not listed here, then simply add the label
+**job-restriction-producer** to your plug-in wiki page and it will be
+automatically listed.
+
+# JIRA issues
+
+If you have any proposals/bug reports, please create an issue on [Jenkins
+JIRA][2].
+
+# Version history
```

Comment:
I would recommend moving it to a separate CHANGELOG.md file

Reply:
Moved.